File: backend/src/LoLCoach.Api/Domain/Player.cs
namespace LoLCoach.Api.Domain;

public sealed class Player
{
    public Guid Id { get; private set; }
    public string Puuid { get; private set; } = null!;
    public string GameName { get; private set; } = null!;
    public string TagLine { get; private set; } = null!;
    public string Region { get; private set; } = null!;
    public DateTimeOffset CreatedAt { get; private set; }
    public DateTimeOffset LastUpdatedAt { get; private set; }

    private Player() { }

    public Player(Guid id, string puuid, string gameName, string tagLine, string region,
        DateTimeOffset createdAt, DateTimeOffset lastUpdatedAt)
    {
        Id = id;
        Puuid = puuid;
        GameName = gameName;
        TagLine = tagLine;
        Region = region;
        CreatedAt = createdAt;
        LastUpdatedAt = lastUpdatedAt;
    }
}

File: frontend/src/app/core/models/player.ts
/** Corpo do POST /api/players/search (spec 001). */
export interface SearchPlayerRequest {
  gameName: string;
  tagLine: string;
  region: string;
}

/** Jogador retornado pelo backend (DTO camelCase). */
export interface Player {
  id: string;
  puuid: string;
  gameName: string;
  tagLine: string;
  region: string;
  createdAt: string;
  lastUpdatedAt: string;
}

/** Problem Details (application/problem+json) do backend. */
export interface ProblemDetails {
  type?: string;
  title?: string;
  status?: number;
  detail?: string;
  traceId?: string;
  /** Validação 400: chaves camelCase -> array de mensagens. */
  errors?: Record<string, string[]>;
}
